modules/model/src/main/scala/lucuma/itc/newTypes.scala

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.itc
+
+import cats.Order
+import cats.syntax.all.*
+import io.circe.Encoder
+import io.circe.Json
+import io.circe.JsonObject
+import io.circe.syntax.*
+import lucuma.core.data.Zipper
+import lucuma.core.data.ZipperCodec.given
+import lucuma.core.util.NewType
+import cats.data.NonEmptyChain
+
+private def customizedZipperEncoder[A: Encoder](
+  allName:      String,
+  selectedName: String, // TODO Eliminate this?
+  indexName:    String
+): Encoder[Zipper[A]] = a =>
+  a.asJson.mapObject(obj =>
+    val map = obj.toMap
+    JsonObject(
+      allName      -> map("all"),
+      selectedName -> map("selected"),
+      indexName    -> map("index")
+    )
+  )
+
+object TargetIntegrationTime extends NewType[Zipper[IntegrationTime]]:
+  extension (a: TargetIntegrationTime)
+    def focusIndex(index: Int): Option[TargetIntegrationTime] =
+      a.value.focusIndex(index).map(TargetIntegrationTime(_))
+
+  given Order[TargetIntegrationTime] = Order.by(_.value.focus)
+
+  given Encoder[TargetIntegrationTime] =
+    customizedZipperEncoder[IntegrationTime]("alternatives", "selected", "selectedIndex")
+      .contramap(_.value)
+type TargetIntegrationTime = TargetIntegrationTime.Type
+
+object AsterismIntegrationTimes extends NewType[Zipper[TargetIntegrationTime]]
+type AsterismIntegrationTimes = AsterismIntegrationTimes.Type
+
+object TargetIntegrationTimeOutcome extends NewType[Either[Error, TargetIntegrationTime]]:
+  given Encoder[TargetIntegrationTimeOutcome] =
+    _.value match
+      case Left(e)  => e.asJson
+      case Right(t) => t.asJson
+type TargetIntegrationTimeOutcome = TargetIntegrationTimeOutcome.Type
+
+object AsterismIntegrationTimeOutcomes extends NewType[NonEmptyChain[TargetIntegrationTimeOutcome]]:
+  extension (a: AsterismIntegrationTimeOutcomes)
+    // If there are errors, return them with their index.
+    // If there are no errors, return an AsterismIntegrationTimes Zipper with the brightest target focused.
+    def partitionErrors: Either[NonEmptyChain[(Error, Int)], AsterismIntegrationTimes] =
+      a.value
+        .traverse(_.value.toOption)
+        .map: nec =>
+          AsterismIntegrationTimes(Zipper.of(nec.head, nec.tail.toList*).focusMin)
+        .toRight(collectErrors.get) // Should be safe to call get here, we know there are errors.
+
+    // The brightest target is the one with the shortest exposure time.
+    // Only returns the index of the brightest target if there are no errors.
+    def brightestIndex: Option[Int] =
```

Comment:
I wonder if `brightest` is the right word here. We are selecting for shortest exposure time which may not be necessarily the brightest (which again can be defined in multiple ways)

Reply:
We can switch to `shortest` or `shortestExpTime`? Other name suggestions?

However, are there situations where we can have the shortest exposure time for one target but that one is not the brightest at the selected wavelength? Maybe for imaging things are different? If this one is actually the brightest, we should call it that, but I'm not sure that is always the case.

Reply:
Maybe `selected`?